File: c/armstrong-numbers/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int candidate)
{
  int originalNumber = candidate;
  int n = 0;
  int result = 0;

  while(originalNumber != 0){
    originalNumber /= 10;
    ++n;
  }

  originalNumber = candidate;
  while(originalNumber != 0) {
    int remainder = originalNumber % 10;
    result += pow(remainder, n);
    originalNumber /= 10;
  }

  return result == candidate;
}

File: zig/armstrong-numbers/armstrong_numbers.zig
const math = @import("std").math;

pub fn isArmstrongNumber(num: u128) bool {
    var original_num: u128 = num;
    var n: u32 = 0;
    var result: u128 = 0;

    while (original_num != 0) : (original_num /= 10) {
        n += 1;
    }

    original_num = num;
    while (original_num != 0) : (original_num /= 10) {
        const remainder: u128 = original_num % 10;
        result += math.pow(u128, remainder, n);
    }

    return result == num;
}
